src/bp/ui-studio/src/web/views/FlowBuilder/diagram/nodes/LinkWidget.tsx

```diff
@@ -9,29 +9,35 @@ class DeletableLinkWidget extends DefaultLinkWidget {
 
   generatePoint(pointIndex: number): JSX.Element {
     const { link } = this.props
-    const x = link.points[pointIndex].x
-    const y = link.points[pointIndex].y
-
-    this.currentLink = link.getID()
+    const point = link.points[pointIndex]
+    const x = point.x
+    const y = point.y
+    const id = point.id
+
+    // When adding a new link
+    if (link.points.length <= 2) {
```

Comment:
Can you move the magic number to a constant somewhere ? This value is also used in diagram/index.tsx file. I suggest something like MAX_NUMBER_OF_POINTS, and also replace the magic number "3" with that

Reply:
The value '2' here and the number '3' from the diagram have two different meaning. '3' represents the number of points you can add to a link (after it has been created) while '2' represents the default number of points in a link (one at the beginning and one at the end). I'll add some clarifications on what those values represent.